pynbody/analysis/halo.py

```diff
@@ -455,6 +455,9 @@ def halo_shape(sim, N=100, rin=None, rout=None, bins='equal', matter_type = 'DM'
         number is not necessarily maintained during fitting. 'log' and 'lin' initialise bins
         with logarithmic and linear radial spacing.
 
+    matter_type : str
```

Comment:
This should be named something like `use_family`, accepting type of either `str` or `pynbody.family.Family`. (See [here](https://pynbody.readthedocs.io/latest/reference/_autosummary/pynbody.family.html) for background on families.)

See below for how to implement this